msi/build/jenkins.wxs

```diff
@@ -0,0 +1,483 @@
+        <Directory Id='JENKINSDIR' Name='$(var.ArtifactName)' FileSource=".">
+          <Component Id='Main' Guid='e4a652bf-c210-4a45-95c4-5dc875b4880b' Win64='yes'>
+            <File Name="jenkins.exe" Source='tmp/jenkins.exe' KeyPath='yes' Id="JenkinsExe"/>
+            <File Id="JenkinsXml" Name="jenkins.xml" Source='tmp/jenkins.xml' DiskId='1' />
+            <File Name="jenkins.exe.config" Source='jenkins.exe.config' DiskId='1' />
+            <File Name="$(var.ArtifactName).war" Source='$(var.WAR)' DiskId='1' />
+            
+            <!-- Update the XML file with the values selected during install -->
+            <util:XmlFile Id='JenkinsXmlJenkinsData' Action='setValue' ElementPath='//service/env/@value' File='[#JenkinsXml]' Value='[JENKINS_ROOT].jenkins' Sequence="1"/>
+            <util:XmlFile Id='JenkinsXmlExecutable' Action='setValue' ElementPath='//service/executable' File='[#JenkinsXml]' Value='[JAVA_HOME]\bin\java.exe' Sequence="1"/>
+            <util:XmlFile Id='JenkinsXmlArguments' Action='setValue' ElementPath='//service/arguments' File='[#JenkinsXml]' Value='-Xrs -Xmx256m -Dhudson.lifecycle=hudson.lifecycle.WindowsServiceLifecycle -jar &quot;[JENKINSDIR]jenkins.war&quot; --httpPort=[PORTNUMBER] --webroot=&quot;[JENKINS_ROOT]war&quot;' Sequence="1"/>
+            <util:XmlFile Id='JenkinsXmlExtensionsPidfile' Action='setValue' ElementPath='//service/extensions/extension/pidfile' File='[#JenkinsXml]' Value='[JENKINS_ROOT]jenkins.pid' Sequence='1'/>
+            <ServiceInstall Id="$(var.ArtifactName)Service"
+                  Name="$(var.ArtifactName)"
+                  DisplayName="$(var.ProductName)"
+                  Type="ownProcess"
+                  Start="auto"
+                  ErrorControl="normal"
+                  Description="$(var.ProductSummary)"
+                  Account="[SERVICE_USERNAME]"
+                  Password="[SERVICE_PASSWORD]" />
+              <ServiceControl Id="Control$(var.ArtifactName)Service" Name="$(var.ArtifactName)" Start="install" Stop="both" Wait="yes" Remove="uninstall"/>
+          </Component>
+
+          <!-- Registry entries -->
+          <Component Id="SettingsRegistryEntries" Guid="">
+              <!-- Do NOT use the application's default registry key here, because THIS key will be removed on uninstall
+                  (important when installing a newer version, because that is uninstall followed by install) -->
+              <RegistryKey Root="HKLM" Key="Software\Jenkins\InstalledProducts\Jenkins">
+                <RegistryValue Name="InstallLocation" Value="[JENKINSDIR]" Type="string" KeyPath="yes" />
+                <RegistryValue Name="Port" Value="[PORTNUMBER]" Type="string" />
+                <RegistryValue Name="SU" Value="[SERVICE_USERNAME]" Type="string" />
+                <RegistryValue Name="SP" Value="[SERVICE_PASSWORD_ENC]" Type="string" />
+                <RegistryValue Name="JenkinsRoot" Value="[JENKINS_ROOT]" Type="string" />
```

Comment:
Why not `JenkinsHome` to be aligned with other occurrences? AFAICT the registry entries are not defined in the previous installer version

Reply:
It's not actually the JENKINS_HOME though, its the root directory that contains other things.